### ISSUE TITLE
Disable auto-insert of imports

### ISSUE THREAD

**Issue by cloudhan** [microsoft/pyright#638]
#207 introduce auto-insert import, it is frustrating, and there seems to be no option to disable it. Please add an option. 

The reason it is frustrating it that pyright never correctly import the needed module or type. It insert some random import at the top of my py file and create an illusion that the type is imported.

**Comment by erictraut** [comment]
Can you provide repro steps for the problem you're seeing? If you're hitting a bug, I'd rather fix the bug that provide a way to disable the feature. Incidentally, it works well for me — I rely on it all the time.

**Comment by cloudhan** [comment]
In some case, actually, it works, but if you are working on some larger or messy project, name clash might be common and pyright will make wrong decision in this case.

For example:

```python
# feature_a.py
class Loader(object):
    pass
```

```python
# feature_b.py
class Loader(object):
    pass
```

when
```
loader = Loader
```
and press tab and line ending will cause a wrong import and there is no way to distinguish at the moment.

In my case, more specifically, I have some variables defined module level, and when writing new code, the name might be suggested by pyright, if I press tab to complete, it might create nasty bug for me and hard to figure out (it is almost silent since the import is inserted to the top of the file).

A proper enhancement will be

allow user to select:
1. auto import
2. confirm suggested import
3. disabled, let me do it on my own completely

**Comment by cloudhan** [comment]
python is a language that is too dynamic to be analyzed statically. Automatically make every decision will eventually accumulate mistakes, and users will only remember the bad part.

Anyway, pyright is a great linter works out of box. It helps me correct many typos and wrong argument on call sites. minimized the edit run crash debug loop. (a.k.a, the Traceback Oriented Programming ;)

**Comment by erictraut** [comment]
The auto-import feature does allow you to choose from among multiple symbols with the same name. In this case, you will see duplicates within the list of completion suggestions, and the module for each one is provided to the right so you can tell them apart. Here's a screen shot that shows two `Logger` symbols, for example.

![Screen Shot 2020-04-24 at 6 52 06 AM](https://user-images.githubusercontent.com/7040122/80220157-4852f000-85f8-11ea-9470-31694e6e3864.png)

Pyright tries to put the most likely replacement first in the list, but it can't always guess correctly. If it does guess incorrectly and you choose the second or third item in the list, it will reorder the list the next time it sees the same completion, putting your previous selection at the top of the list.

**Comment by erictraut** [comment]
I'm closing this for now. If you have specific examples of where the functionality isn't working as designed, please provide repro cases, and I'll work on fixing them.

**Comment by udnaan** [comment]
I have a case where a string is inserted, quoted, bracketed and all and the `feature` auto inserted an imported which was similarly named but not quite exact module. 

The string was 'environemnt.m.53' and the module was simply `environment`.

Can you provide the option to disable the feature as I am picky about what gets imported anyway. An error is better than magic that can cause much grief later.

**Comment by FugueSoft** [comment]
Just got started with Zed and this has gotta be the worst thing I've encountered so far.
Dropping this into my `settings.json` does not seem to do the trick unfortunately

```
  "lsp": {
    "pyright":{
      "settings":{
        "python.analysis":{
          "autoImportCompletions": false
        }
      }
    }
```

**Comment by FugueSoft** [comment]
Alright so I don't know what I did but it seems to be working now

**Comment by devesnosy** [comment]
> Just got started with Zed and this has gotta be the worst thing I've encountered so far. Dropping this into my `settings.json` does not seem to do the trick unfortunately
> 
> ```
>   "lsp": {
>     "pyright":{
>       "settings":{
>         "python.analysis":{
>           "autoImportCompletions": false
>         }
>       }
>     }
> ```
> 
> Example Video: https://github.com/user-attachments/assets/1ec72d30-8678-49ba-8112-26c9c962fc62
> 
> E: dropped it in my `pyright.json` and pyright still yields the same result E2: It doesn't seem to be a problem with untracked files as new files in the directory don't have this issue.
> 
> However, as soon as I `git add` the file, the issue comes right back

same problem here Zed somehow doesn't respect the setting :/

```
// Zed settings
//
// For information on how to configure Zed, see the Zed
// documentation: https://zed.dev/docs/configuring-zed
//
// To see all of Zed's default settings without changing your
// custom settings, run `zed: open default settings` from the
// command palette (cmd-shift-p / ctrl-shift-p)
{
  "ui_font_size": 16,
  "buffer_font_size": null,
  "theme": {
    "mode": "system",
    "light": "One Light",
    "dark": "Andromeda"
  },
  "ui_font_family": "Noto Mono",
  "buffer_font_family": "Noto Mono",
  "extend_comment_on_newline": false,
  "lsp": {
    "pyright": {
      "settings": {
        "python.analysis": {
          "diagnosticMode": "workspace",
          "typeCheckingMode": "strict",
          "autoImportCompletions": false
        },
        "python": {
          "pythonPath": ".venv/bin/python"
        }
      }
    }
  }
}

```
